pangeo_forge_recipes/transforms.py

```diff
@@ -647,22 +611,37 @@ class StoreToZarr(beam.PTransform, ZarrWriterMixin):
       based on the full dataset (e.g. divide along a certain dimension based on a desired chunk
       size in memory). For more advanced chunking strategies, check
       out https://github.com/jbusecke/dynamic_chunks
-    :param dynamic_chunking_fn_kwargs: Optional keyword arguments for ``dynamic_chunking_fn``.
     :param attrs: Extra group-level attributes to inject into the dataset.
     :param encoding: Dictionary encoding for xarray.to_zarr().
     :param append_dim: Optional name of the dimension to append to.
+
+    Example of using a wrapper function to reduce the arity of a more complex dynamic_chunking_fn:
+
+        Suppose there's a function `calculate_dynamic_chunks` that requires extra parameters: an
+        `xarray.Dataset`, a `target_chunk_size` in bytes, and a `dim_name` along which to chunk.
+        To fit the expected signature for `dynamic_chunking_fn`, we can define a wrapper function
+        that presets `target_chunk_size` and `dim_name`:
+
+        ```python
+        def calculate_dynamic_chunks(ds, target_chunk_size, dim_name) -> Dict[str, int]:
+            ...
+
+        def dynamic_chunking_wrapper(ds: xarray.Dataset) -> Dict[str, int]:
+            target_chunk_size = 1024 * 1024 * 10
+            dim_name = 'time'
+            return calculate_dynamic_chunks(ds, target_chunk_size, dim_name)
+
+        StoreToZarr(..., dynamic_chunking_fn=dynamic_chunking_wrapper, ...)
+        ```
     """
 
-    # TODO: make it so we don't have to explicitly specify combine_dims
-    # Could be inferred from the pattern instead
     combine_dims: List[Dimension]
     store_name: str
     target_root: Union[str, FSSpecTarget, RequiredAtRuntimeDefault] = field(
         default_factory=RequiredAtRuntimeDefault
     )
     target_chunks: Dict[str, int] = field(default_factory=dict)
     dynamic_chunking_fn: Optional[Callable[[xr.Dataset], dict]] = None
-    dynamic_chunking_fn_kwargs: Optional[dict] = field(default_factory=dict)
```

Comment:
Is there an advantage to lower arity here aside from an aesthetically tidier signature?

Reply:
The tidy signature is much of the benefit I see but also hopefully passing responsibility downstream to pipeline writers/users in a way that can't hide any magic we might be applying. On this conception of things, the user has one thing to worry about: a function that produces the chunking they desire given a template dataset